Source/CheckoutAPIClient.swift

```diff
@@ -134,7 +152,13 @@ public class CheckoutAPIClient {
         urlRequest.httpBody = try? jsonEncoder.encode(card)
 
         request(urlRequest)
-            .validate().responseJSON { response in
+            .validate().responseJSON { [weak self] response in
+
+                guard let strongSelf = self else {
+
+                    completion(.failure(.objectDeallocatedUnexpectedly))
+                    return
+                }
```

Comment:
just want to double check, the places where we're using weak and strong self are weak wherever possible right?

Reply:
Yep, I updated the "new" methods but not the deprecated methods as there is no way to exit/return a `NetworkError` (only `ErrorResponse`), e.g. 

```
@available(*, deprecated, message: "Use createApplePayToken with a Swift Result type for completion handler.")
    public func createApplePayToken(paymentData: Data,
                                    successHandler: @escaping (CkoCardTokenResponse) -> Void,
                                    errorHandler: @escaping (ErrorResponse) -> Void)
```